### PR TITLE
fix: (Core|Platform) fix optional rtl service

### PR DESCRIPTION
**Pull request by Betrozov**
#### Please provide a link to the associated issue.
Fixes https://github.com/SAP/fundamental-ngx/issues/4802
#### Please provide a brief summary of this pull request.
Fixed optional rtl service
#### Please check whether the PR fulfills the following requirements

- [X] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [X] tests for the changes that have been done
- [X] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [X] Run npm run build-pack-library and test in external application

Documentation checklist:
- [X] update `README.md`
- [X] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [X] Documentation Examples
- [X] Stackblitz works for all examples

### CODE REVIEW

**Review by olshansky — APPROVED** [review]

File: libs/core/src/lib/slider/slider.component.ts [hunk 1]
```diff
@@ -713,9 +713,13 @@ export class SliderComponent implements OnInit, OnChanges, OnDestroy, ControlVal
 
     /** @hidden Rtl change subscription */
     private _subscribeToRtl(): void {
-        this._isRtl = this._rtlService?.rtl.getValue();
+        if (!this._rtlService) {
+            return;
+        }
+
+        this._isRtl = this._rtlService.rtl.getValue();
```
Comment (dimamarksman) [inline libs/core/src/lib/slider/slider.component.ts:720]: `this._rtlService.rtl` is a behavior subject so there is no need for this operation

File: libs/platform/src/lib/components/split-menu-button/split-menu-button.component.ts [hunk 2]
```diff
@@ -96,10 +96,12 @@ export class SplitMenuButtonComponent extends BaseComponent implements OnInit, A
         this.secondaryId = 'secondary-' + this.id;
         // if no title provided.
         this.title = this.title || this.buttonLabel;
-        this._rtlChangeSubscription = this._rtl.rtl.subscribe((isRtl: boolean) => {
-            this.dir = isRtl ? 'rtl' : 'ltr';
-            this._cd.detectChanges();
-        });
+        if (this._rtl) {
+            this._rtlChangeSubscription = this._rtl.rtl.subscribe((isRtl: boolean) => {
```
Comment (dimamarksman) [inline libs/platform/src/lib/components/split-menu-button/split-menu-button.component.ts:100]: `this._rtl.rtl` looks weird

File: libs/core/src/lib/pagination/pagination.component.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ import { KeyUtil } from '../utils/functions';
 import { PaginationService } from './pagination.service';
 import { RtlService } from '../utils/services/rtl.service';
 import { Pagination } from './pagination.model';
+import { Subscription } from 'rxjs';
```
Comment (dimamarksman) [inline libs/core/src/lib/pagination/pagination.component.ts:24]: Please keep right order of imports

**Review by dimamarksman — APPROVED** [review]
LGTM,
added a few comments

**Review by KevinOkamoto — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Review by Seamoo13 — APPROVED** [review]
tested RTL for the affected components, looks good.

**Pull request merged**